### ISSUE TITLE
POST Error thrown from network stack causes ConcurrentModificationException

### ISSUE THREAD

**Issue by vsayer** [one-signal/OneSignal-Android-SDK#23]
This issue was caught developing the Xamarin SDK which builds on top of OneSignal-Android-SDK 2.1.1.

The issue appears to be occurring after DeleteTags.

DeleteTags was called to clean up any tags on the OneSignal server. It looks like the DeleteTags call did succeed, but a low-level Java exception was thrown in the process:
```java
[libc] getaddrinfo called from pid =19367
[libc] getaddrinfo called from pid =19367
[libc] dnsproxy getaddrinfo returns 0
[OneSignal] POST Error thrown from network stack. 
[OneSignal] java.util.ConcurrentModificationException
[OneSignal] 	at java.util.HashMap$HashIterator.nextEntry(HashMap.java:806)
[OneSignal] 	at java.util.HashMap$KeyIterator.next(HashMap.java:833)
[OneSignal] 	at com.onesignal.OneSignalStateSynchronizer$UserState.persistStateAfterSync(OneSignalStateSynchronizer.java:313)
[OneSignal] 	at com.onesignal.OneSignalStateSynchronizer$UserState.access$600(OneSignalStateSynchronizer.java:150)
[OneSignal] 	at com.onesignal.OneSignalStateSynchronizer$2.onSuccess(OneSignalStateSynchronizer.java:454)
[OneSignal] 	at com.onesignal.OneSignalRestClient.makeRequest(OneSignalRestClient.java:117)
[OneSignal] 	at com.onesignal.OneSignalRestClient.postSync(OneSignalRestClient.java:76)
[OneSignal] 	at com.onesignal.OneSignalStateSynchronizer.syncUserState(OneSignalStateSynchronizer.java:436)
[OneSignal] 	at com.onesignal.OneSignalStateSynchronizer$NetworkHandlerThread$1.run(OneSignalStateSynchronizer.java:361)
[OneSignal] 	at android.os.Handler.handleCallback(Handler.java:733)
[OneSignal] 	at android.os.Handler.dispatchMessage(Handler.java:95)
[OneSignal] 	at android.os.Looper.loop(Looper.java:136)
[OneSignal] 	at android.os.HandlerThread.run(HandlerThread.java:61)
[OneSignal] Failed last request. statusCode: 200
```

**Comment by vsayer** [comment]
So it appears to be pretty deterministic. The first time you install your Android app, the issue doesn't occur, but if you install it again and call the DeleteTags, it consistently occurs. The only way to not see the exception is uninstall the application entirely from the device.

**Comment by jkasten2** [comment]
Fixed in pull request #24.
